### PR TITLE
Change aria for active pagination nav control

### LINKED ISSUE #960: [Global] Active pagination nav control issues
## Issue summary

The active pagination item features `aria-*` attributes which may not be well supported on the current element. This could lead to a confusing user experience when assistive technology is unable to accurately convey the purpose or state of the element.

In addition, the visual treatment of the active, yet inactive control features an underline. This could confuse sighted users into thinking the control is available to be activated.

<details>
  <summary>Screenshots</summary>

  Dawn glob post. Underlined, inactive, pagination control is highlighted.

</details>

#### HTML

```html
<span
  class="pagination__item pagination__item--current"
  aria-current="page"
  aria-label="Page 1"
>
  1
</span>
```

#### CSS

```css
.pagination__item--current::after,
.pagination__item:hover::after {
  content: '';
  display: block;
  width: 2rem;
  position: absolute;
  bottom: 8px;
  left: 50%;
  transform: translateX(-50%);
  background-color: currentColor;
}
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a collection page or blog page with many comments
3. Run [axe DevTools](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd) and visually inspect the active control

### Expected

1. `aria-*` to be applied to supported elements.
2. A subdued visual indicating an inactive control.

### Actual

1. `aria-*` attributes are applied to low-supported elements.
2. The inactive control appears actionable.

## Recommendation

1. Swap the `span` element for an `a`.
2. Add the `role` attribute, setting its value to `link` – helps to share semantic meaning.
3. Add the `aria-disabled` attribute, setting its value to `true` – helps to share element state.
4. Adjust the visual affordance of the control by removing the underline and adding the `light` class. Use the `[role="link"]:not([href])` selector for additional visual styling.

#### HTML

```html
<a
  class="pagination__item pagination__item--current light"
  aria-current="page"
  aria-label="Page 1"
  role="link"
  aria-disabled="true"
>
  1
</a>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.4.0
- **Component:** Global
- **WCAG Principal:** Perceivable, Robust
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** Low
- **Effort:** Low

### PR DESCRIPTION
**Pull request by pizzaz93**
#960

This change might need to be applied to both the preceding and following lines.

### CODE REVIEW

File: snippets/pagination.liquid [hunk 1]
```diff
@@ -32,7 +32,7 @@
             <a href="{{ part.url }}{{ anchor }}" class="pagination__item link" aria-label="{{ 'general.pagination.page' | t: number: part.title }}">{{ part.title }}</a>
           {%- else -%}
             {%- if part.title == paginate.current_page -%}
-              <span class="pagination__item pagination__item--current" aria-current="page" aria-label="{{ 'general.pagination.page' | t: number: part.title }}">{{ part.title }}</span>
+              <a class="pagination__item pagination__item--current light" aria-current="page" aria-label="{{ 'general.pagination.page' | t: number: part.title }}" role="link" aria-disabled="true">{{ part.title }}</a>
```
Comment (ludoboludo) [inline snippets/pagination.liquid:35]:
```suggestion
              <a role="link" aria-disabled="true" class="pagination__item pagination__item--current light" aria-current="page" aria-label="{{ 'general.pagination.page' | t: number: part.title }}">{{ part.title }}</a>
```

**Review by ludoboludo — COMMENTED** [review]
This isn't taking care of the underline styling of the current page @svinkle mentioned but I wonder if it's really necessary to update it. 

I feel like it's a visual help 🤔 
If it's something that needs to happen, then maybe we don't close the original issue so that some designers can explore what different styling could be applied instead.

**Comment by pizzaz93** [comment]
> This isn't taking care of the underline styling of the current page @svinkle mentioned but I wonder if it's really necessary to update it.
> 
> I feel like it's a visual help 🤔 If it's something that needs to happen, then maybe we don't close the original issue so that some designers can explore what different styling could be applied instead.

So removing the fixes from the PR comment and just having this PR mentioned in that issue?

**Comment by ludoboludo** [comment]
> So removing the fixes from the PR comment and just having this PR mentioned in that issue?

Yeah exactly 👍

**Review by LucasLacerdaUX — APPROVED** [review]

**Comment by ludoboludo** [comment]
Thanks @pizzaz93 👍 🚀

**Pull request merged**